emailCrawler.java

```diff
@@ -31,14 +24,13 @@
 
 	public static void main(String args[]) throws IOException {
 		String url = "http://" + args[0];
-		// this list is used to store all of urls which can be discovered in
-		// home page. Actually, I can write this program
-		// by using DFS so that I can find more discoverable pages on discovered
-		// pages. But I don't do it for that it may go into a infinite loop.
+		// this list is used to store all of urls which can be discovered in home page. Actually, I can write this 
+		// program by using DFS so that I can find more discoverable pages on pages which we have discovered. 
+		// But here, I don't do it for that it may go into a infinite loop.
```

Comment:
So as this is coded, only the links on the page resolved by the initial argument URL will be searched for emails.  If that argument page does not all the links on the site map you could potentially miss some.
How can you make sure you crawl all the pages of the site without getting into the infinite loop you mentioned?